docs/tutorials/reward_networks.rst

```diff
@@ -0,0 +1,131 @@
+
+    from imitation.rewards.reward_nets import RewardNet
+    import torch as th
+
+    class MyRewardNet(RewardNet):
+        def __init__(self, observation_space, action_space):
+            super().__init__(observation_space, action_space)
+            # initialize your custom reward network here
+
+        def forward(self,
+            state: th.Tensor, # (batch_size, *obs_shape)
+            action: th.Tensor, # (batch_size, *action_shape)
+            next_state: th.Tensor, # (batch_size, *obs_shape)
+            done: th.Tensor, # (batch_size,)
+        ) -> th.Tensor:
+            # implement your custom reward network here
+            return th.zeros_like(done) # (batch_size,)
+
+Replace an Environment's Reward with a Reward Network
+-----------------------------------------------------
+
+In order to use a reward network to train a policy, we need to integrate it into an environment. This is done by wrapping the environment in a :class:`RewardVecEnvWrapper <imitation.rewards.reward_wrapper.RewardVecEnvWrapper>`. This wrapper replaces the environment's reward function with the reward network's function.
+
+.. testsetup::
+    :skipif: skip_doctests
+
+    import numpy as np
+    rng = np.random.default_rng(0)
+    from gym.spaces import Box
+    obs_space = Box(np.ones(2), np.ones(2))
+    action_space = Box(np.ones(5), np.ones(5))
+
+.. testcode::
+    :skipif: skip_doctests
+
+    from imitation.util import util
+    from imitation.rewards.reward_wrapper import RewardVecEnvWrapper
+    from imitation.rewards.reward_nets import BasicRewardNet
+
+    reward_net = BasicRewardNet(obs_space, action_space)
+    venv = util.make_vec_env("Pendulum-v1", n_envs=3, rng=rng)
+    venv = RewardVecEnvWrapper(venv, reward_net.predict_processed)
+
+Reward Network Wrappers
+-----------------------
+
+Imitation learning algorithms should converge to a reward function that will theoretically induce the optimal/soft optimal policy. However, these reward functions may not always be well suited for training RL agents, or we may want to modify them to encourage exploration, for instance.
+
+There are two types of wrapper:
+* :class:`ForwardWrapper <imitation.rewards.reward_nets.ForwardWrapper>` allows for direct modification of the results of the reward network's ``forward`` method. It is used during the learning of the reward network and thus must be differentiable. These wrappers are always applied first and are always active. They are used for applying transformations like potential shaping (see :class:`ShapedRewardNet <imitating.rewards.reward_nets.ShapedRewardNet>`).
+* :class:`PredictProcessedWrapper <imitation.rewards.reward_nets.PredictProcessedWrapper>` modifies the predict_processed call to the reward network. Thus this type of reward network wrapper is designed to only modify the reward when it is being used to train/evaluate a policy but *not* when we are taking gradients on it. Thus it does not have to be differentiable.
+
```

Comment:
Does not render correctly in sphinx. Try dashes and indentation an extra newline?

What doe you mean by "always active"? Why should they not be always active?